python_apps/airtime-celery/airtime-celery/tasks.py

```diff
@@ -0,0 +1,53 @@
+import os
+import json
+import urllib2
+import soundcloud
+from celery import Celery
+from celery.utils.log import get_task_logger
+
+celery = Celery()
+logger = get_task_logger(__name__)
+
+
+@celery.task(name='soundcloud-upload', acks_late=True)
+def soundcloud_upload(data, token, file_path):
+    """
+    Upload a file to SoundCloud
+
+    :param data:      associative array containing SoundCloud metadata
+    :param token:     OAuth2 client access token
+    :param file_path: path to the file being uploaded
+
+    :return: the SoundCloud response object
+    :rtype: dict
+    """
+    client = soundcloud.Client(access_token=token)
+    # Open the file with urllib2 if it's a cloud file
+    data['asset_data'] = open(file_path, 'rb') if os.path.isfile(file_path) else urllib2.urlopen(file_path)
```

Comment:
This task should be able to download the track from Airtime using the media REST API. There's a single download URL that should work regardless of where the track is. (It should always take a URL, not a path.)

Also, use requests, not urllib2. There's a bunch of gotchas to using urllib2, like that there's no timeout by default. If the host doesn't reply from that urlopen, this task will hang forever.